0xPolygon/polygon-edge
# Add discard reason in eth_getTransactionReceipt

# Description

This PR lets `eth_getTransactionReceipt` return message indicating the reason of tx discard.

```json
{
  "id": 1,
  "jsonrpc": "2.0",
  "method": "eth_getTransactionReceipt",
  "params": [
    "0x...." 
  ]
}
```

If the reason is recorded in journal, JSON-RPC returns string reason as an error instead of nil.

```json
{
  "message": "Tx discarded by reaching recover limit"
}
```

# Changes include


# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it

# Checklist


## Testing


### Manual tests

1. Setup & Start Edge clusters in the usual manner
2. Send a transaction
3. Call `eth_txStatus` giving the tx hash and make sure the status is right

# Documentation update

Please link the documentation update PR in this section if it's present, otherwise delete it

# Additional comments

Please post additional comments in this section if you have them, otherwise delete it
